### PR TITLE
Fix test db path on Windows

### PR DESCRIPTION
**Pull request by Tobbe**
My take on https://github.com/redwoodjs/redwood/pull/854

### CODE REVIEW

**Comment by Tobbe** [comment]
Leaving this PR as a draft for now, because I have not been able to manually test it yet. I build my changes and copy them over with `yarn rwt copy ../redwood` but then when I try to run my tests I get this

```
tobbe@XPS9550 MINGW64 ~/dev/redwood/v14
$ yarn rw test
yarn run v1.22.4
$ C:\Users\tobbe\dev\redwood\v14\node_modules\.bin\rw test
internal/modules/cjs/loader.js:1032
  throw err;
  ^

Error: Cannot find module 'lodash-decorators'
Require stack:
- C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\structure\dist\x\decorators.js
- C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\structure\dist\ide.js
- C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\structure\dist\index.js
- C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\cli\dist\commands\diagnostics.js
- C:\Users\tobbe\dev\redwood\v14\node_modules\yargs\index.js
- C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\cli\dist\index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1029:15)
    at Function.Module._load (internal/modules/cjs/loader.js:898:27)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
    at Object.<anonymous> (C:\Users\tobbe\dev\redwood\v14\node_modules\@redwoodjs\structure\dist\x\decorators.js:32:25)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\@redwoodjs\\structure\\dist\\x\\decorators.js',
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\@redwoodjs\\structure\\dist\\ide.js',
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\@redwoodjs\\structure\\dist\\index.js',
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\@redwoodjs\\cli\\dist\\commands\\diagnostics.js',
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\yargs\\index.js',
    'C:\\Users\\tobbe\\dev\\redwood\\v14\\node_modules\\@redwoodjs\\cli\\dist\\index.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Tobbe** [comment]
Fixed the missing module error by running `yarn install` inside `node_modules/@redwoodjs/structure`

The DB path works in git-bash now

**Comment by Tobbe** [comment]
@RobertBroersma Can you take a look at this please?

File: packages/internal/src/paths.ts [hunk 1]
```diff
@@ -193,3 +193,15 @@ export const processPagesDir = (
   })
   return deps
 }
+
+export const toPosixPathOnWindows = (path: string) => {
```
Comment (Tobbe) [inline packages/internal/src/paths.ts:197]: If anyone has a better name for this function I'm all ears 👂

File: packages/cli/src/commands/test.js [hunk 2]
```diff
@@ -10,7 +11,7 @@ const jest = require('jest')
 // TODO: Get from redwood.toml
 const sides = getProject().sides
 
-// https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/test.js#L39
+// https://github.com/facebook/create-react-app/blob/cbad256a4aacfc3084be7ccf91aad87899c63564/packages/react-scripts/scripts/test.js#L39
```
Comment (Tobbe) [inline packages/cli/src/commands/test.js:14]: This actually has nothing to do with this PR, but it felt silly to create a separate PR just for this. At least I made a separate commit for it...

**Comment by RobertBroersma** [comment]
Works on my machine and the checks are passing. Nice one!

**Comment by peterp** [comment]
There's `URL_file` in structure that might actually do the exact same thing:

https://github.com/redwoodjs/redwood/blob/20f3dc93c4e7f99edf6e31a606ce158301543678/packages/structure/src/x/URL.ts#L10-L15

**Comment by Tobbe** [comment]
I tried using that `URL_file` function. But it doesn't work. At least not for my use-case.

Here's the output

```
$ yarn rw test
yarn run v1.22.4
$ C:\Users\tobbe\dev\redwood\redwood-testing\node_modules\.bin\rw test
cacheDirDb file:/c/Users/tobbe/dev/redwood/redwood-testing/node_modules/.redwood/test.db
urlFileName file:///C:/Users/tobbe/dev/redwood/redwood-testing/node_modules/.redwood/test.db
$ C:\Users\tobbe\dev\redwood\redwood-testing\node_modules\.bin\rw db up
Migrate database up... [started]
$ C:\Users\tobbe\dev\redwood\redwood-testing\node_modules\.bin\prisma migrate up --experimental --create-db
Error: Command failed with exit code 255: C:\Users\tobbe\dev\redwood\redwood-testing\node_modules\@prisma\cli\migration-engine-windows.exe cli --datasource file:///C:/Users/tobbe/dev/redwood/redwood-testing/node_modules/.redwood/test.db create-database
Jul 21 19:02:47.919  INFO migration_engine: Starting migration engine CLI git_hash="45c4da4dd3ccd6a322796b228bdf937c7ce884e8"
Jul 21 19:02:47.919 ERROR migration_engine::commands: Creating SQLite database parent directory.

{"is_panic":false,"message":"Creating SQLite database parent directory.\n","backtrace":null}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Migrate database up... [failed]
→ Command failed with ENOENT: yarn prisma migrate up --experimental --create-db
spawn yarn prisma ENOENT
Command failed with ENOENT: yarn prisma migrate up --experimental --create-db
spawn yarn prisma ENOENT
```

`cacheDirDb` is the path "my" code generates, and using that works.
`urlFileName` is what I get using `URL_file`, and that doesn't work.

This is how I use `URL_file`

```
    const urlFileName = URL_file(`${CACHE_DIR}/test.db`)
    console.log('urlFileName', urlFileName)
    const DATABASE_URL = process.env.TEST_DATABASE_URL || urlFileName
```

**Comment by peterp** [comment]
Ah ok, thanks for looking into that for me :)

File: packages/internal/src/paths.ts (hunk 1 above)
Comment (peterp) [reply packages/internal/src/paths.ts:201]: Make something like `ensurePosixPath`

**Comment by peterp** [comment]
LGTM!

**Pull request merged**